File: interview_cake/trees_and_graphs/binary_tree.rb
class TreeNode
  attr_accessor :value, :left, :right
  def initialize(value)
    @value = value
    @left, @right = nil, nil
  end
end

ROOT = TreeNode.new(1)
ROOT.left = TreeNode.new(2)
ROOT.left.left = TreeNode.new(3)
ROOT.left.right = TreeNode.new(4)
ROOT.right = TreeNode.new(5)
ROOT.right.left = TreeNode.new(6)
ROOT.right.right = TreeNode.new(7)

File: interview_cake/trees_and_graphs/binary_tree.js
class TreeNode {
  constructor(value) {
    this.value = value;
    this.right = null;
    this.left = null;
  }
}

export let root = new TreeNode(1);
root.left = new TreeNode(2);
root.left.left = new TreeNode(3);
root.left.right = new TreeNode(4);
root.right = new TreeNode(5);
root.right.left = new TreeNode(6);
root.right.right = new TreeNode(7);
root.right.right.right = new TreeNode(8);
root.right.right.right.right = new TreeNode(9);
root.right.right.right.rightleft = new TreeNode(10);

